testing/sd_stress/sd_stress_main.c

```diff
@@ -0,0 +1,404 @@
+{
+  printf("Stress test on a mount point\n");
+  printf(CONFIG_TESTING_SD_STRESS_PROGNAME ": [-r] [-b] [-f]\n");
+  printf("  -r   Number of runs (%u-%u), default %u\n",
+                              min_runs, max_runs, default_runs);
+  printf("  -b   Number of bytes (%u-%u), default %u\n",
+                              min_bytes, max_bytes, default_bytes);
+  printf("  -f   Number of files (%u-%u), default %u\n",
+                              min_files, max_files, default_files);
+}
+
+static bool create_dir(const char *path)
+{
+  int ret = mkdir(TEMPDIR, S_IRWXU | S_IRWXG | S_IRWXO);
+
+  if (ret < 0)
+    {
+      printf("mkdir %s failed, ret: %d, errno: %d -> %s\n",
+             path, ret, errno, strerror(errno));
+      return false;
+    }
+
+  return true;
+}
+
+static bool remove_dir(const char *path)
+{
+  int ret = rmdir(TEMPDIR2);
+
+  if (ret < 0)
+    {
+      printf("rmdir %s failed, ret: %d, errno: %d -> %s\n",
+             path, ret, errno, strerror(errno));
+      return false;
+    }
+
+  return true;
+}
+
+static bool create_files(const char *dir, const char *name,
+                         size_t num_files, char *bytes, size_t num_bytes)
+{
+  bool passed = true;
+  if (num_files > 999)
+    {
+      printf("too many files\n");
+      return false;
+    }
+
+  char *read_bytes = (char *)malloc(num_bytes);
+
+  if (!read_bytes)
+    {
+      printf("malloc failed for read bytes bufffer\n");
+      return false;
+    }
+
+  size_t path_len = strlen(dir) + strlen(name);
+
+  if (path_len + 5 >= MAX_PATH_LEN)
+    {
+      printf("path name too long\n");
+      return false;
+    }
+
+  for (size_t i = 0; i < num_files; ++i)
+    {
+      char path[MAX_PATH_LEN];
+      snprintf(path, MAX_PATH_LEN, "%s/%s%03u", dir, name, i);
+
+      memset(read_bytes, 0x0, num_bytes);
+
+      /* Fill the file with a set of incrementing bytes */
+
+      for (size_t j = 0; j < num_bytes; j++)
+        {
+          bytes[j] = (bytes[j] + i) & 0xff;
+        }
+
+      int fd = open(path, O_CREAT | O_RDWR);
+
+      if (fd < 0)
+        {
+          printf("open %s failed, errno: %d -> %s\n",
+                 path, errno, strerror(errno));
+          passed =  false;
+          break;
+        }
+
+      int ret = write(fd, bytes, num_bytes);
+
+      if (ret != (int)num_bytes)
+        {
+          printf("write %s failed, ret: %d, errno %d -> %s\n",
+                 path, ret, errno, strerror(errno));
+          passed =  false;
+          break;
+        }
+
+      ret = lseek(fd, 0, SEEK_SET);
+
+      if (ret < 0)
+        {
+          printf("lseek %s failed, ret: %d, errno %d -> %s\n",
+                 path, ret, errno, strerror(errno));
+          passed =  false;
+          break;
+        }
+
+      ret = read(fd, read_bytes, num_bytes);
+
+      if (ret != (int)num_bytes)
+        {
+          printf("read %s failed, ret: %d, errno %d -> %s\n",
+                 path, ret, errno, strerror(errno));
+          passed =  false;
```

Comment:
```suggestion
          passed = false;
```
please fix all place